src/deploy.ts

```diff
@@ -49,77 +49,114 @@ export interface DeployedContracts {
 export async function deployAll(
   signer: ethers.Signer
 ): Promise<DeployedContracts> {
-  const gov = await deployGovernance(signer);
-  const rad = await deployRad(signer);
+  const signerAddr = await signer.getAddress();
+  const rad = await deployRadicleToken(signer, signerAddr);
+  const timelock = await deployTimelock(signer, signerAddr, 2 * 60 * 60 * 24);
+  const gov = await deployGovernance(
+    signer,
+    timelock.address,
+    rad.address,
+    signerAddr
+  );
   const exchange = await deployExchange(rad, signer);
-  const ens = (await deployContract(signer, ENSRegistry, [])) as ENS;
-  const registrar = await deployRegistrar(exchange, ens, signer);
+  const ens = await deployEns(signer);
+  await setSubnodeOwner(ens, "", "eth", signerAddr);
+  const registrar = await deployRegistrar(
+    signer,
+    await exchange.oracle(),
+    exchange.address,
+    rad.address,
+    ens,
+    "eth",
+    "radicle",
+    signerAddr
+  );
   const ethPool = await deployEthPool(signer, 10);
   const erc20Pool = await deployErc20Pool(signer, 10, rad.address);
 
   return { gov, rad, exchange, registrar, ens, ethPool, erc20Pool };
 }
 
-export async function deployRad(signer: ethers.Signer): Promise<RadicleToken> {
-  const signerAddr = await signer.getAddress();
-  const radToken = await new RadicleToken__factory(signer).deploy(signerAddr);
+export async function deployRadicleToken(
+  signer: ethers.Signer,
+  account: string
+): Promise<RadicleToken> {
+  return await new RadicleToken__factory(signer).deploy(account);
+}
 
-  return radToken;
+export async function deployVestingToken(
+  signer: ethers.Signer,
+  token: string,
+  owner: string,
+  beneficiary: string,
+  amount: ethers.BigNumberish,
+  vestingStartTime: ethers.BigNumberish,
+  vestingPeriod: ethers.BigNumberish,
+  cliffPeriod: ethers.BigNumberish
+): Promise<VestingToken> {
+  return await new VestingToken__factory(signer).deploy(
+    token,
+    owner,
+    beneficiary,
+    amount,
+    vestingStartTime,
+    vestingPeriod,
+    cliffPeriod
+  );
 }
 
 export async function deployRegistrar(
-  exchange: Exchange,
-  ens: ENS,
-  signer: ethers.Signer
+  signer: ethers.Signer,
+  oracle: string,
+  exchange: string,
+  token: string,
+  ens: ENS, // The connected signer must be the owner of the domain
+  domain: string,
+  label: string,
+  admin: string
 ): Promise<Registrar> {
-  const signerAddr = await signer.getAddress();
-  const oracle = await exchange.oracle();
-  const rad = await exchange.rad();
   const registrar = await new Registrar__factory(signer).deploy(
     ens.address,
-    ensUtils.nameHash("eth"),
-    ensUtils.labelHash("radicle"),
+    ensUtils.nameHash(label + "." + domain),
     oracle,
-    exchange.address,
-    rad,
-    signerAddr
+    exchange,
+    token,
+    admin
   );
+  await setSubnodeOwner(ens, domain, label, registrar.address);
```

Comment:
This will only work if we deploy using the owner of `radicle.eth` :thinking: 

Maybe this line should be removed, and only be used in the test-ens scenario? Because in the real deployment, we will probably manually transfer ownership, or perhaps it could be done in a separate deployment step where the owner can connect their wallet?

Reply:
I've fixed it, but haven't pushed the changes, please take a look again

Reply:
Looks good.